### PR TITLE
Add MariaDB metrics configurations

### PR DESCRIPTION
**Pull request by pkbhowmick**
Signed-off-by: Pulak Kanti Bhowmick <pulak@appscode.com>

### CODE REVIEW

File: charts/kubedb-crds/templates/mariadb-metrics-config.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+      type: gauge
+      labels:
+        - key: unit
+          value: bytes
+      params:
+        - key: memory
+          valuePath: .spec.podTemplate.spec.resources.limits.memory
+      metricValue:
+        valueFromExpression: calculateMemory(memory)
+
```
Comment (spectro30) [inline charts/kubedb-crds/templates/mariadb-metrics-config.yaml:165]: maybe add a newline at end of file.

File: charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      field: 
+        path: .metadata.creationTimestamp
+        type: DateTime
+      metricValue:
+        valueFromPath: .metadata.creationTimestamp
+
+    - name: kubedb_mariadb_version_info
+      help: "MariaDB versions info"
+      type: gauge
+      labels:
```
Comment (spectro30) [inline charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml:22]: how about adding the ```deprecated``` field?

File: charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+          valuePath: .spec.db.image
+        - key: exporter_image
+          valuePath: .spec.exporter.image
+        - key: stash_addon_backup_task
+          valuePath: .spec.stash.addon.backupTask.name
+        - key: stash_addon_restore_task
+          valuePath: .spec.stash.addon.restoreTask.name
+      metricValue: 
+        value: 1
+
```
Comment (spectro30) [inline charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml:33]: maybe add a newline at end of file.

**Review by spectro30 — CHANGES_REQUESTED** [review]

**Review by spectro30 — DISMISSED** [review]
LGTM!

File: charts/kubedb-crds/templates/mariadb-metrics-config.yaml [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+              valueFromExpression: "toInt(phase == 'NotReady')"
+          - labelValue: DataRestoring
+            metricValue:
+              valueFromExpression: "toInt(phase == 'DataRestoring')"
+
+    - name: kubedb_mariadb_info
+      help: "MariaDB instance information"
+      type: gauge
+      labels:
+        - key: sslMode
```
[the remark this answers is not in the document]
Comment (pkbhowmick) [reply charts/kubedb-crds/templates/mariadb-metrics-config.yaml]: Added

File: charts/kubedb-crds/templates/mariadb-metrics-config.yaml [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+          valuePath: .spec.sslMode
```
[the remark this answers is not in the document]
Comment (pkbhowmick) [reply charts/kubedb-crds/templates/mariadb-metrics-config.yaml]: Added

File: charts/kubedb-crds/templates/mariadb-metrics-config.yaml (hunk 1 above)
Comment (pkbhowmick) [reply charts/kubedb-crds/templates/mariadb-metrics-config.yaml]: Added

File: charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml (hunk 2 above)
Comment (pkbhowmick) [reply charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml:22]: Added

File: charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml (hunk 3 above)
Comment (pkbhowmick) [reply charts/kubedb-crds/templates/mariadb-versions-metrics-config.yaml]: Added

**Review by spectro30 — DISMISSED** [review]

**Pull request merged**